TeslaMateAgile/Services/BarryService.cs

```diff
@@ -0,0 +1,87 @@
+﻿using Microsoft.Extensions.Options;
+using System.Globalization;
+using System.Text.Json.Serialization;
+using TeslaMateAgile.Data;
+using TeslaMateAgile.Data.Options;
+using TeslaMateAgile.Services.Interfaces;
+
+namespace TeslaMateAgile.Services;
+
+public class BarryService : IPriceDataService
+{
+    private readonly HttpClient _client;
+    private readonly BarryOptions _options;
+
+    public BarryService(HttpClient client, IOptions<BarryOptions> options)
+    {
+        _client = client;
+        _options = options.Value;
+    }
+
+    public async Task<IEnumerable<Price>> GetPriceData(DateTimeOffset from, DateTimeOffset to)
+    {
+        var request = new BarryRequest {
+            Params = new string[] { _options.MPID, from.UtcDateTime.ToString("o"), to.UtcDateTime.ToString("o")}.ToList()
+        };
+        var objAsJson = System.Text.Json.JsonSerializer.Serialize<BarryRequest>(request);
+        var content = new StringContent(objAsJson, System.Text.Encoding.UTF8, "application/json");
+
+        var resp = await _client.PostAsync(_options.BaseUrl, content);
+        resp.EnsureSuccessStatusCode();
+
+        var barryResponse = await System.Text.Json.JsonSerializer.DeserializeAsync<BarryResponse>(await resp.Content.ReadAsStreamAsync());
```

Comment:
Name can be simplified

```suggestion
        var barryResponse = await JsonSerializer.DeserializeAsync<BarryResponse>(await resp.Content.ReadAsStreamAsync());
```